### PR TITLE
Add .gomplateignore support

### PR DESCRIPTION
**Pull request by zealic**
For #218

### CODE REVIEW

**Comment by hairyhenderson** [comment]
Thanks for this, @zealic! When I have a little more time I'll review this more fully, but in the meantime can you add an integration test? Have a look at the existing `.bats` files in `test/integration/` for inspiration. You can run the tests with `test-integration-docker`. I'm mostly looking for automated confirmation that the compiled binary processes `.gomplateignore` files as expected.

**Comment by zealic** [comment]
OK, I will try add integration tests. @hairyhenderson

**Comment by zealic** [comment]
Integration tests added, please review and merge.

File: Gopkg.toml [hunk 1]
```diff
@@ -14,3 +14,7 @@
 [[constraint]]
   branch = "v2"
   name = "gopkg.in/hairyhenderson/yaml.v2"
+
+[[constraint]]
+  name = "github.com/zealic/xignore"
+  version = "0.3.2"
```
Comment (hairyhenderson) [inline Gopkg.toml]: is there a reason you're pinning to this release? this shouldn't be necessary in this case - there's a tag and `dep` will understand it just fine...

File: docs/content/usage.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ You can specify multiple `--file` and `--out` arguments. The same number of each
 
 For processing multiple templates in a directory you can use `--input-dir` and `--output-dir` together. In this case all files in input directory will be processed as templates and the resulting files stored in `--output-dir`. The output directory will be created if it does not exist and the directory structure of the input directory will be preserved.  
 
+You can use `.gomplateignore` ignore some files, have the similar behavior to the [.gitignore](https://git-scm.com/docs/gitignore) file.
```
Comment (hairyhenderson) [inline docs/content/usage.md]:
```suggestion
You can use `.gomplateignore` to ignore some files in the input directory, with similar syntax and behaviour to [.gitignore](https://git-scm.com/docs/gitignore) files.
```

**Comment by hairyhenderson** [comment]
Thanks for updating this, @zealic! I'll be able to take a deeper look in a few hours, hopefully.

File: Gopkg.toml (hunk 1 above)
Comment (zealic) [reply Gopkg.toml]: I think minor range version `~` is better than fixed version.
I will maintain upstream repo to keep stability

I will maintain upstream repo to keep stablelity.

## See also:
* https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md#version
* https://semver.org

File: template.go [hunk 3]
```diff
@@ -14,8 +14,12 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/spf13/afero"
+	"github.com/zealic/xignore"
 )
 
+// Gomplateignore ignore file name, like .gitignore
+const Gomplateignore = ".gomplateignore"
```
Comment (hairyhenderson) [inline template.go]: Let's leave this unexported for now - I don't see a reason to export this yet:

```suggestion
const gomplateignore = ".gomplateignore"
```

(obviously references below will need to change)

**Comment by hairyhenderson** [comment]
Thanks @zealic! And, thanks especially for your persistence in getting this in - I know it took a very long time! 😉

**Pull request merged**